torch/csrc/cuda/nccl.cpp

```diff
@@ -0,0 +1,212 @@
+#include "nccl.h"
+#include "torch/csrc/THP.h"
+#include "torch/csrc/Types.h"
+#include "torch/csrc/cuda/THCP.h"
+
+#include <nccl.h>
+#include <sstream>
+#include <unordered_map>
+
+#ifdef ncclGroupStart
```

Comment:
Uh is there no other way to get NCCL version? @ngimel 
This is a bit fragile, because if this will change to be a function at some point, we'll silently compile for NCCL 1.

Reply:
this was wrong, i fixed it. it also fixed the NCCL2 getting stuck / deadlocked.